app/components/user_item/user_item.tsx

```diff
@@ -121,6 +121,10 @@ const UserItem = ({
     const shared = user ? isShared(user) : false;
     const deactivated = user ? isDeactivated(user) : false;
 
+    if (deactivated) {
+        return null;
+    }
```

Comment:
Wouldn't this cause all users deactivated to no longer be displayed?  Could you help me clarify that it will not affect other places vs Create Direct Message screen?

Also, take a look at this [chunk of code](https://github.com/mattermost/mattermost-mobile/blob/main/app/screens/create_direct_message/create_direct_message.tsx#L250):

```
        if (restrictDirectMessage) {
            results = await searchProfiles(serverUrl, lowerCasedTerm, {team_id: currentTeamId, allow_inactive: true});
        } else {
            results = await searchProfiles(serverUrl, lowerCasedTerm, {allow_inactive: true});
        }
```

@larkox you've made some changes on the chunk of code above, would it be the right way to only show active users?

Reply:
You are right, @rahimrahman . Removing the `allow_inactive` should return us the list of users without deactivated users. That seems to be the right way to do this in this case.

0/5 on whether this will create any problem, like when trying to access old dms with deactivated users, but maybe we can access that in a different way.

Reply:
@rahimrahman @larkox thank you for the feedback will adjust this accordingly

Reply:
Thank you all folks for your persistence and collaboration to make it happen, - so much looking forward to the fix here from [8074](https://github.com/mattermost/mattermost-mobile/issues/8074)